#165273579 : User can favorite article

nadralia opened the pull request:
#### What does this PR do?

Allow an authenticated user to favourite and unfavourite an article.
Prevent an unauthenticated user from favouriting and unfavouriting an article.
#### How should this be manually tested?
Make migrations and run migrate
`python manage.py makemigrations`
`python manage.py migrate`

Run the application:
`python manage.py runserver`
Open postman
Register a user using the route
http://127.0.0.1:8000/api/users/
Login the user you just registered
http://127.0.0.1:8000/api/users/login/
Get the token that was returned
Insert a header in postman with a key of Authorization and value Bearer <token>
Then visit http://127.0.0.1:8000/api/articles/ to create an article using POST.
Then visit  http://127.0.0.1:8000/api/articles/<slug>/favorite to either favourite  using POST or unfavourite an article DELETE.

#### What are the relevant pivotal tracker stories?

 [#165273579](https://www.pivotaltracker.com/story/show/165273579)

ZackAtama commented on authors/apps/articles/views.py:
```diff
@@ -82,4 +93,72 @@ def put(cls, request, slug):
+                {
+                    'message': "You have already unfavourited this article"
+                }, status.HTTP_400_BAD_REQUEST)
+
```
Hello @nadralia , great work and logic, however, for a user who has never `favorited` an article, a proper error message is needed: instead of `'message': "You have already unfavourited this article"`, let the message be close to `"message":"You have not favourited this article"`.

kalsmic replied:
@nadralia, Besides this, the other logic in your code looks good to me

nadralia replied:
Fixed it

llwasampijja reviewed (approved):
Good work @nadralia Once you work on that small change that @AtamaZack pointed out, you should be good to go!

bisonlou reviewed (approved):
This looks good @nadralia. Well done

nadralia committed:
ft(Favorite): User can favorite article

add post favorite endpoint
add delete favorite endpoint
write tests for post and delete endpoints

[Finishes #165273579]

llwasampijja reviewed (changes requested):
@nadralia, resolve the merge conflicts and you will be good to go!

Done

kalsmic reviewed (approved):
well done @nadralia
